### PR TITLE
Fix / Bitmart Connector

### PR DESCRIPTION
**Pull request by superlazycoder**
**Before submitting this PR, please make sure**:

- [x] Your code builds clean without any errors or warnings
- [x] You are using approved title ("feat/", "fix/", "docs/", "refactor/")

**A description of the changes proposed in the pull request**:

added string check for json loads with warning output for failure
changed spot/depth from 400 to 50
added decompress and url to user stream data source
skipped missing orders in update order status

**Tests performed by the developer**:
Live trading using Pure Market Making strategy

**Tips for QA testing**:
Now that it is functioning, the maintainer will likely need to finish the parts that I likely missed. 
I dont know anything about the HK urls.
Its possible my changes merely solved the issues that I experienced and could have introduced other issues. I have been using hummingbot for a little over a week, and did this repair overnight. Hopefully its a good enough starting point to get the process moved along.

### CODE REVIEW

**Comment by assinnata** [comment]
thanks, @superlazycoder I tested the PR and it is working fine

**Comment by superlazycoder** [comment]
If anyone wants to check into the change, and see if the jobs finish this time.

File: hummingbot/connector/exchange/bitmart/bitmart_exchange.py [hunk 1]
```diff
@@ -641,6 +641,9 @@ async def _update_order_status(self):
             responses = await safe_gather(*tasks, return_exceptions=True)
             for response in responses:
                 if isinstance(response, Exception):
+                    if hasattr(response, "message"):
+                        if response.message == "order not exist":
```
Comment (mhrvth) [inline hummingbot/connector/exchange/bitmart/bitmart_exchange.py:645]: Is this a real message? I can't find it in the Bitmart API documentation.

File: hummingbot/connector/exchange/bitmart/bitmart_api_order_book_data_source.py [hunk 2]
```diff
@@ -222,7 +228,7 @@ async def listen_for_order_book_diffs(self, ev_loop: asyncio.BaseEventLoop, outp
                 for trading_pair in self._trading_pairs:
                     ws_message: WSRequest = WSRequest({
                         "op": "subscribe",
-                        "args": [f"spot/depth400:{convert_to_exchange_trading_pair(trading_pair)}"]
+                        "args": [f"spot/depth50:{convert_to_exchange_trading_pair(trading_pair)}"]
```
Comment (mhrvth) [inline hummingbot/connector/exchange/bitmart/bitmart_api_order_book_data_source.py:231]: What's the reason for lowering depth?

**Review by mhrvth — CHANGES_REQUESTED** [review]
-- Reviewed in representation of Hummingbot Foundation Technical DAO

File: hummingbot/connector/exchange/bitmart/bitmart_exchange.py (hunk 1 above)
Comment (superlazycoder) [reply hummingbot/connector/exchange/bitmart/bitmart_exchange.py:645]: Its a message I saw when trying to remove and/or retrieve the order. Skipping removing items that didnt exist seemed legitimate to me.

File: hummingbot/connector/exchange/bitmart/bitmart_api_order_book_data_source.py (hunk 2 above)
Comment (superlazycoder) [reply hummingbot/connector/exchange/bitmart/bitmart_api_order_book_data_source.py:231]: Its what is available. 

image

**Review by mhrvth — APPROVED** [review]
Thanks for the replies. LGTM then.

**Comment by carlitogetaladajr** [comment]
PRP at Snapshot:

https://snapshot.org/#/hbot-prp.eth/proposal/0x8ae60e50253ad2ea68b5156efe60b9b4e1a62ed8b4ee08719791220b2f102a32

File: hummingbot/connector/exchange/bitmart/bitmart_api_order_book_data_source.py (hunk 2 above)
Comment (aarmoa) [reply hummingbot/connector/exchange/bitmart/bitmart_api_order_book_data_source.py:231]: Is the channel `spot/depth400` really invalid or is it just not documented? If it is not invalid we should use it anyway, even if it is not documented.

Comment (superlazycoder) [reply hummingbot/connector/exchange/bitmart/bitmart_api_order_book_data_source.py:231]: I'll tell you what, That was one of the first fixes I put in for errors. Its possible other things made that an error. 
I'll test it out with 400 again, and see what happens.

Comment (superlazycoder) [reply hummingbot/connector/exchange/bitmart/bitmart_api_order_book_data_source.py]: There do not appear to be any adverse effects to the depth400. 
Someone else should test it first however.

Comment (mhrvth) [reply hummingbot/connector/exchange/bitmart/bitmart_api_order_book_data_source.py]: What depth is it actually receiving? Really the 400?

Comment (superlazycoder) [reply hummingbot/connector/exchange/bitmart/bitmart_api_order_book_data_source.py]: I am not sure. My client asked me to get something for bitmart, fixing the errors was the most logical pathway given the time constraints. At this point this change is back to what it used to be and is not erroring. So this change (from documentation) was not necessary to fix the bugs. 

Hopefully the repair either gets added in, or becomes a good place for the maintainer to continue their development on the connector.

**Comment by aarmoa** [comment]
I tried to register to the channel `spot/depth400` in my local environment and it fails. I think we should explore the alternative of updating the connector instead of fixing a couple of issues. It is evident now that the connector in its current state is not notifying the errors properly and makes the strategies think everything is ok when the subscriptions are failing.

**Comment by JeremyKono** [comment]
Closing this PR in favor of [#5400](https://github.com/hummingbot/hummingbot/pull/5400)

**Pull request closed without merging**